cmake/Modules/FindComputeCpp.cmake

```diff
@@ -275,12 +293,14 @@ function(add_sycl_to_target targetName binaryDir sourceFiles)
   )
   # Add custom target to run compute++ and generate the integration header
   foreach(sourceFile ${sourceFiles})
+    message (STATUS "binaryDir = ${binaryDir}")
     __build_spir(${targetName} ${sourceFile} ${binaryDir} ${fileCounter})
     MATH(EXPR fileCounter "${fileCounter} + 1")
   endforeach()
 
   # Link with the ComputeCpp runtime library
-  target_link_libraries(${targetName} PUBLIC ${COMPUTECPP_RUNTIME_LIBRARY}
-                        PUBLIC ${OpenCL_LIBRARIES})
+  target_link_libraries(${targetName} PUBLIC $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:${COMPUTECPP_RUNTIME_LIBRARY_DEBUG}>
```

Comment:
I realise this solves the problem rather neatly but I'd prefer for the logic to be factored out, put into a different variable and then referenced in the link_libraries call - this isn't very readable!

Reply:
I am not sure it can be factored out due to the nature of generator expressions in CMake...

Reply:
OK, that's fair. I guess what I'm saying is: I won't ever be able to change that, probably :smiley:

Reply:
It can be changed to ordinary script logic at the cost of LoC. I wanted it to remain concise. I do agree the generator expression syntax is not the nicest, but it is less LoC. Will change it to `if-else` if you wish.

Reply:
If it can't be moved, fair enough. In that case just leave it. It's unlikely to need to change, so I'll just have to learn to love it!